packages/ui/src/ui/pages/navigation/content/Document/DocumentEditModal.tsx

```diff
@@ -0,0 +1,61 @@
+import React, {FC, useRef, useState} from 'react';
+import {Dialog} from '@gravity-ui/uikit';
+import {
+    EditJsonProps,
+    EditJsonWithPreview,
+} from '../../../../components/EditJsonWithPreview/EditJsonWithPreview';
+import cn from 'bem-cn-lite';
+import './DocumentEditModal.scss';
+
+type Props = {
+    open: boolean;
+    saving: boolean;
+    document: any;
+    onCancel: () => void;
+    onSave: (data: string) => void;
+};
+
+const block = cn('document-edit-modal');
+
+const DocumentEditModal: FC<Props> = ({open, document, onCancel, saving, onSave}) => {
+    const [error, setError] = useState(false);
+    const result = useRef<string>(JSON.stringify(document, null, '\t'));
+
+    const handleChange = (data: EditJsonProps['value']) => {
+        result.current = data.value || '';
+        setError(Boolean(data.error));
+    };
+
+    const handleSaveDocument = () => {
+        onSave(result.current);
+    };
+
+    return (
+        <Dialog onClose={onCancel} open={open} modalClassName={block()}>
```

Comment:
We have special component to generate forms, it is better to use it here:
```ts
const [visible, setVisible] = React.useState(false);
...
return (
     <YTDFDIalog
          visible={visible}
          onClose={() => setVisible(false)}
          onAdd={() => {...}}
          fields=[{name: 'value', type: 'json', caption: 'Caption'}]
     />
```
And we don't need to remember `editing`/`saving` state in the redux store, so the following actions are unnecessary: `SET_DOCUMENT_SAVING`, `SET_DOCUMENT_EDIT_MODE`.